### PR TITLE
Bug 1921610: fix pipeline metrics font size issue

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/ODC-5312

**Analysis / Root cause**: 

Font size in the chart labels should be consistent while resizing the window to a smaller screen.

**Solution Description**: 

Used resize Observer (react measure) to pass the container width to PF charts.

**Screen shots / Gifs for design review**: 

metrics-fontsize

**Test setup:**
1. Run pipelines multiple times and visit pipeline Metrics page.
2. Resize the metrics page to smaller size.

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

/kind bug
cc: @andrewballantyne @christianvogt @bgliwa01

### CODE REVIEW

**Review by bgliwa01 — APPROVED** [review]
lgtm! Thanks, Karthik!

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineRunTaskRunGraph.tsx [hunk 1]
```diff
@@ -115,43 +115,47 @@ const PipelineRunTaskRunGraph: React.FC<PipelineMetricsGraphProps> = ({
...
+          <LineChart
+            ariaDesc={t('pipelines-plugin~Pipeline task run duration chart')}
+            ariaTitle={t('pipelines-plugin~Task run duration')}
+            data={_.values(finalObj)}
+            yTickFormatter={(seconds) => getYaxisValues(seconds)}
+            events={getEvents()}
+            hiddenSeries={hiddenSeries}
+            tickValues={tickValues}
+            width={contentRect.bounds.width}
+            height={chartHeight}
```
Comment (invincibleJai) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineRunTaskRunGraph.tsx:130]: can we even get height from `contentRect.bounds.height`

Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineRunTaskRunGraph.tsx:130]: This is the last chart in the bottom row and this one has interactive legends under the chart, whereas the other chart in the bottom row doesn't have this legends. So fixed Height will make the cards even, if i use a dynamic height the cards wont be on the same size.

**Comment by invincibleJai** [comment]
/approve

Verified the changes , works as expected

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/__tests__/PipelineRunCount.spec.tsx [hunk 2]
```diff
@@ -58,6 +58,8 @@ describe('Pipeline Run Count Graph', () => {
   it('Should render an TimeSeriesChart if data is available', () => {
     pipelineRunSpy.mockReturnValue([{ data: { result: [{ x: 'x' }] } }, false, false]);
     const PipelineRunCountWrapper = shallow(<PipelineRunCount {...PipelineRunCountProps} />);
-    expect(PipelineRunCountWrapper.find(TimeSeriesChart).exists()).toBe(true);
+    expect(PipelineRunCountWrapper.find(LoadingInline).exists()).toBe(false);
+    expect(PipelineRunCountWrapper.find(GraphEmpty).exists()).toBe(false);
+    expect(PipelineRunCountWrapper.find(Measure).exists()).toBe(true);
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/__tests__/PipelineRunCount.spec.tsx:63]: @karthikjeeyar Is this change really necessary? The change of this tests "... if data is available" doesn't check that there is a TimeSeriesChart. I would like to see that this test case still tests for the TimeSeriesChart, or? Wdyt?

**Review by jerolimov — CHANGES_REQUESTED** [review]

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/__tests__/PipelineRunCount.spec.tsx (hunk 2 above)
Comment (jerolimov) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/__tests__/PipelineRunCount.spec.tsx:63]: @karthikjeeyar looks like you need to go two levels deeper with the shallow-renderer to find your TimeSeriesChart again. One for the `<Measure />` and one for the children-render-function:

```
    expect(
      PipelineRunCountWrapper.dive()
        .dive()
        .find(TimeSeriesChart)
        .exists(),
    ).toBe(true);
```

Want add this test again? Up to you if you want also keep the other checks. With the Measure check you test some internal which are not required in my opionion.

Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/__tests__/PipelineRunCount.spec.tsx:63]: Agreed @jerolimov. I will update the tests and push again. Thanks for pointing out.

Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/__tests__/PipelineRunCount.spec.tsx]: Updated.

**Review by jerolimov — APPROVED** [review]
Thanks for the test update.

/lgtm

**Comment by spadgett** [comment]
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.
